pkg/system_api.go

```diff
@@ -162,7 +162,9 @@ type Component struct {
 	CostTotal7d     float64                     `json:"cost_total_7d,omitempty" gorm:"column:cost_total_7d"`
 	CostTotal30d    float64                     `json:"cost_total_30d,omitempty" gorm:"column:cost_total_30d"`
 	LogSelectors    dutyTypes.LogSelectors      `json:"logs,omitempty" gorm:"column:log_selectors"`
-	ParentLookup    *v1.ParentLookup            `json:"parentLookup,omitempty" gorm:"-"`
+
+	ParentLookup *v1.ParentLookup `json:"parentLookup,omitempty" gorm:"-"`
+	StatusExpr   string           `json:"statusExpr,omitempty" gorm:"-"`
```

Comment:
Doesn't this need to be  saved on the db, so it can be run again on query ?

Reply:
That is persisted on the topology spec.

Reply:
https://github.com/flanksource/duty/pull/876